tools/distpartitioning/gloo_wrapper.py

```diff
@@ -59,6 +59,40 @@ def alltoall_cpu(rank, world_size, output_tensor_list, input_tensor_list):
     for i in range(world_size):
         dist.scatter(output_tensor_list[i], input_tensor_list if i == rank else [], src=i)
 
+def alltoall_cpu_object_lst(rank, world_size, input_list):
+    """
+    Each process scatters list of input objects to all processes in a cluster
+    and return gathered list of objects in output list. 
+
+    Parameters
+    ----------
+    rank : int
+        The rank of current worker
+    world_size : int
+        The size of the entire
+    input_tensor_list : List of tensor
+        The tensors to exchange
+
+    Returns
+    -------
+    list: list of objects are received from other processes
+       This is the list of objects which are sent to the current process by
+       other processes as part of this exchange
+    """
+    rcv_list = []
+    output_list = [None] * world_size
+    for i in range(world_size):
```

Comment:
is it right how the for loop works as follow:
* in iteration i, the input data is scattered to all processes from process i;
* in other iterations, process i receives one object in each iteration.
pytorch distributed doesn't support an API like that. this is basically all processes scatter its data to all processes.

Reply:
Correct. Basically this code is just an adaptation of the sample code present on the pytorch website at the following location (https://pytorch.org/docs/stable/distributed.html). This is adapted from the all_to_all sample implementation as mentioned here.